src/test/java/io/debezium/connector/yugabytedb/YugabyteDBPublicationReplicationTest.java

```diff
@@ -0,0 +1,214 @@
+
+    @BeforeAll
+    public static void beforeClass() throws SQLException {
+        initializeYBContainer();
+        TestHelper.dropAllSchemas();
+    }
+
+    @BeforeEach
+    public void before() throws Exception {
+        initializeConnectorTestFramework();
+        TestHelper.executeDDL("yugabyte_create_tables.ddl");
+    }
+
+    @AfterEach
+    public void after() throws Exception {
+        stopConnector();
+        Awaitility.await()
+            .atMost(Duration.ofSeconds(65))
+            .until(() -> {
+                try {
+                    return dropReplicationSlot();
+                } catch (Exception e) {
+                    return false;
+                }
+            });
+        TestHelper.execute(TestHelper.dropPublicationStatement);
+        TestHelper.executeDDL("drop_tables_and_databases.ddl");
+    }
+
+    @AfterAll
+    public static void afterClass() {
+        shutdownYBContainer();
+    }
+
+    @Test
+    public void testPublicationReplicationStreamingConsumption() throws Exception {
+        TestHelper.execute(String.format(TestHelper.createPublicationForTableStatement, "pub", "t1"));
+        TestHelper.execute(TestHelper.createReplicationSlotStatement);
+
+        Configuration.Builder configBuilder = TestHelper.getConfigBuilderWithPublication("yugabyte", "pub", "test_replication_slot"); 
+        startEngine(configBuilder);
+        final long recordsCount = 1;
+
+        awaitUntilConnectorIsReady();
+
+        Thread.sleep(30000);
+
+        insertRecords(recordsCount);
+
+        verifyPrimaryKeyOnly(recordsCount);
+    }
+
+    @Test
+    public void testPublicationReplicationSnapshotConsumption() throws Exception {
+        String insertStatement = "INSERT INTO t2 values (%d);";
+        final int recordsCount = 1000;
+        TestHelper.executeBulk(insertStatement, recordsCount);
+
+        TestHelper.execute(String.format(TestHelper.createPublicationForTableStatement, "pub", "t2"));
+        TestHelper.execute(TestHelper.createReplicationSlotStatement);
+
+        Configuration.Builder configBuilder = TestHelper.getConfigBuilderWithPublication("yugabyte", "pub", "test_replication_slot"); 
+        configBuilder.with(YugabyteDBConnectorConfig.SNAPSHOT_MODE, YugabyteDBConnectorConfig.SnapshotMode.INITIAL.getValue());
+        startEngine(configBuilder);
+
+        awaitUntilConnectorIsReady();
+
+        verifyRecordCount(recordsCount);
+    }
+
+    @Test
+    public void testAllTablesPublicationAutoCreateMode() throws Exception {
+        TestHelper.execute(TestHelper.createReplicationSlotStatement);
+
+        Configuration.Builder configBuilder = TestHelper.getConfigBuilderWithPublication("yugabyte", "pub", "test_replication_slot");
+        configBuilder.with(YugabyteDBConnectorConfig.PUBLICATION_AUTOCREATE_MODE, "all tables");
+        startEngine(configBuilder);
+        final int recordsCount = 10;
+
+        awaitUntilConnectorIsReady();
+
+        Thread.sleep(30000);
```

Comment:
Why the sleep?

Reply:
This is to ensure that we insert the records after bootstrap is completed. Adding a code comment for this.